### ISSUE TITLE
Linear view: interpolation to flight track altitude?

### ISSUE THREAD

**Issue by gisi90** [Open-MSS/MSS#2409]
Does linear view interpolate to the flight altitude of the flight track? 
I just realized this is not the case for our WMS. 
![grafik](https://github.com/Open-MSS/MSS/assets/54586339/6e476726-181f-4714-b762-5610e8ed12ce)

**Comment by ReimarBauer** [comment]
it behaves on the campaigns2019 server on a first few better.  Needs be detailed checked.

![image](https://github.com/Open-MSS/MSS/assets/278728/e78d3e6d-de55-4c57-8930-a30bcf79fb84)

**Comment by ReimarBauer** [comment]
lets look on temperature on the demodata, example.ftml

```
<?xml version="1.0" encoding="utf-8"?>
  <FlightTrack version="9.0.0">
    <ListOfWaypoints>
      <Waypoint flightlevel="233.0" lat="41.601070573320186" location="" lon="41.355120439498535">
        <Comments></Comments>
      </Waypoint>
      <Waypoint flightlevel="374.0" lat="48.19354838709677" location="" lon="33.74841526975632">
        <Comments></Comments>
      </Waypoint>
      <Waypoint flightlevel="372.0" lat="51.23623045499366" location="" lon="-34.20481757994082">
        <Comments></Comments>
      </Waypoint>
      <Waypoint flightlevel="20.0" lat="37.037047471474835" location="" lon="-40.797295393717434">
        <Comments></Comments>
      </Waypoint>
    </ListOfWaypoints>
  </FlightTrack>

```
point 3 is wrong

![image](https://github.com/Open-MSS/MSS/assets/278728/4aad4775-3c10-4ee6-8795-9e0c4ef29e0f)

**Comment by joernu76** [comment]
The interpolation done in  mslib/mswms/mss_plot_driver.py looks a bit fishy on how different vertical coordinate axis types are handled.

Could you dump the header of the file containing the interpolated data and the configuration of the employed linear view layer?

**Comment by ReimarBauer** [comment]
for the demodata it is:
20121017_12_ecmwf_forecast.T.EUR_LL015.036.ml.nc

```
netcdf \20121017_12_ecmwf_forecast.T.EUR_LL015.036.ml {
dimensions:
        time = 7 ;
        hybrid = 18 ;
        lat = 40 ;
        lon = 100 ;
variables:
        float time(time) ;
                time:units = "hours since 2012-10-17T12:00:00.000Z" ;
                time:standard_name = "time" ;
        float hybrid(hybrid) ;
                hybrid:standard_name = "atmosphere_hybrid_sigma_pressure_coordinate" ;
                hybrid:units = "sigma" ;
                hybrid:positive = "down" ;
                hybrid:formula = "p(time,level,lat,lon) = ap(level) + b(level) * ps(time,lat,lon)" ;
                hybrid:formula_terms = "ap: hyam b: hybm ps: Surface_pressure_surface" ;
        float hyam(hybrid) ;
                hyam:units = "Pa" ;
                hyam:standard_name = "atmosphere_hybrid_pressure_coordinate" ;
        float hybm(hybrid) ;
                hybm:units = "1" ;
                hybm:standard_name = "atmosphere_hybrid_height_coordinate" ;
        float lat(lat) ;
                lat:units = "degrees_north" ;
                lat:standard_name = "latitude" ;
        float lon(lon) ;
                lon:units = "degrees_north" ;
                lon:standard_name = "longitude" ;
        float air_temperature(time, hybrid, lat, lon) ;
                air_temperature:standard_name = "air_temperature" ;
                air_temperature:units = "K" ;
                air_temperature:grid_mapping = "LatLon_Projection" ;
                air_temperature:missing_value = NaNf ;

```
it is the default demodata setup

` (mpl_lsec_styles.LS_DefaultStyle, "air_potential_temperature", ["ecmwf_EUR_LL015"]),`

**Comment by joernu76** [comment]
The current code had a lot of issues. I can see some of them could relate to your problem. There is an PR to stable that fixes these  issues.

**Comment by gisi90** [comment]
fix #2425 did the trick. 
Before: 
![grafik](https://github.com/Open-MSS/MSS/assets/54586339/24044e4a-9148-44a3-aa2d-e85b548d652a)
After including the changes:
![grafik](https://github.com/Open-MSS/MSS/assets/54586339/9bc95994-b5bf-4cf9-aafb-58cfeac53aea)
